expan/core/experiment.py

```diff
@@ -306,49 +306,37 @@ def _quantile_filtering(self, data, kpis, percentile, threshold_type):
             flags = flags | data[column].apply(method_table[threshold_type])
         return flags
 
-
-    def chi_square_test_result_and_statistics(self, variant_column, weights, min_counts=5, alpha=0.05):
-        """ Tests the consistency of variant split with the hypothesized distribution.
-        
-        :param variant_column: variant column from the input data frame
-        :param weights: dict with variant names as keys, weights as values
-                        ({<variant_name>:<weight>, ...}
-        :param min_counts: minimum number of observed and expected frequencies (should be at least 5), see 
-                            http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html
-        :param alpha: significance level, 0.05 by default
-        :return: True(if split is consistent with the given split) or
-                 False(if split is not consistent with the given split)
-        :rtype:  Boolean, float, float, pd.Series, pd.Series
+    def run_goodness_of_fit_test(self, observed_freqs, expected_freqs, alpha=0.01, min_counts=5):
+        """ Checks the validity of observed and expected counts and runs chi-square test for goodness of fit.
+
+        :param observed_freqs: observed frequencies
+        :type  observed_freqs: pd.Series
+        :param expected_freqs: expected frequencies
+        :type  expected_freqs: pd.Series
+        :param alpha: significance level
+        :type  alpha: float
+        :param min_counts: minimum number of observations to run chi-square test
+        :type  min_counts: int
+        :return split_is_unbiased: False is split is biased and True if split is correct
+        :rtype: bool
         """
-        if not hasattr(variant_column, '__len__'):
-            raise ValueError("Variant split check was cancelled since input variant column is empty or doesn't exist.")
-        if not hasattr(weights, '__len__'):
-            raise ValueError("Variant split check was cancelled since input weights are empty or doesn't exist.")
-        if len(weights) <= 1 or len(variant_column) <= 1:
-            raise ValueError("Variant split check was cancelled since input weights or the number if categories "
-                             "is less than 2.")
 
-        # Count number of observations per each variant
-        variant_column = pd.Series(variant_column).dropna(axis=0)
-        observed_freqs = variant_column.value_counts()
+        if not hasattr(observed_freqs, '__len__'):
+            raise ValueError("Variant split check was cancelled since input observed frequencies are empty "
```

Comment:
hmm... why not get the length and compare to 0? if it doesn't have ```'__len__'```, the input type is wrong I guess...